### PR TITLE
UserService: use the UserService instead of calling sqlstore directly

### PR DESCRIPTION
**Pull request by mildwonkey**
This PR refactors various calls to the sqlstore user-related methods with calls to the user service.

There is one major change hiding in this PR. UserService.Delete originally called a number of services to delete user-related records. I moved everything except the actual call to the user table, and moved those into the API. This was done to avoid dependencies cycles; many of our services depend on the user service, so the user service itself should have as few dependencies as possible.

### CODE REVIEW

**Comment by ying-jeanne** [comment]
one last thing we might need to pay attention is that all the services call was ramped into WithTransactionalDbSession, so it was belong to the same transaction, when you move it to the api, we need to make sure it is still in the same transaction. (ensured by sharing ctx)

**Comment by mildwonkey** [comment]
@ying-jeanne : true! that's exactly what's made this whole project so interesting. 

(we've already talked about this in meetings; transcribing my thoughts here for posterity)  
We can't split grafana into services without splitting up certain transactions, and until we settle on a distributed transaction method (built here or not) we will inevitably have some behavioral changes such as this one. 

I'm not as worried about deletions right now; dangling records aren't the biggest problem we could face, and we know that distributed transactions (or more deliberate behavior changes) are needed eventually.

File: pkg/services/user/userimpl/store.go [hunk 1]
```diff
@@ -23,6 +25,17 @@ type store interface {
 type sqlStore struct {
 	db      db.DB
 	dialect migrator.Dialect
+	logger  log.Logger
+	cfg     *setting.Cfg
+}
+
+func ProvideStore(db db.DB, cfg *setting.Cfg) sqlStore {
```
[the remark this answers is not in the document]
Comment (idafurjes) [reply pkg/services/user/userimpl/store.go:32]: oh yes, I remember

**Review by idafurjes — APPROVED** [review]
Great work :)

**Pull request merged**